internal/bminventory/inventory.go

```diff
@@ -1553,27 +1553,83 @@ func (b *bareMetalInventory) GetClusterSupportedPlatforms(ctx context.Context, p
 
 func (b *bareMetalInventory) UpdateClusterInstallConfigInternal(ctx context.Context, params installer.V2UpdateClusterInstallConfigParams) (*common.Cluster, error) {
 	log := logutil.FromContext(ctx, b.log)
-	var cluster common.Cluster
+	var cluster *common.Cluster
+	var err error
 	query := "id = ?"
 
-	err := b.db.First(&cluster, query, params.ClusterID).Error
-	if err != nil {
+	txSuccess := false
+	tx := b.db.Begin()
+	defer func() {
+		if !txSuccess {
+			log.Error("UpdateClusterInstallConfigInternal failed")
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			log.Error("UpdateClusterInstallConfigInternal failed to recover")
+			tx.Rollback()
+		}
+	}()
+
+	if cluster, err = common.GetClusterFromDBForUpdate(tx, params.ClusterID, common.UseEagerLoading); err != nil {
 		log.WithError(err).Errorf("failed to find cluster %s", params.ClusterID)
 		return nil, err
 	}
 
-	if err = b.installConfigBuilder.ValidateInstallConfigPatch(&cluster, params.InstallConfigParams); err != nil {
+	if err = b.installConfigBuilder.ValidateInstallConfigPatch(cluster, params.InstallConfigParams); err != nil {
 		return nil, common.NewApiError(http.StatusBadRequest, err)
 	}
 
-	err = b.db.Model(&common.Cluster{}).Where(query, params.ClusterID).Update("install_config_overrides", params.InstallConfigParams).Error
+	// Set install config overrides feature usage
+	err = b.setInstallConfigOverridesUsage(cluster.Cluster.FeatureUsage, params.InstallConfigParams, params.ClusterID, tx)
 	if err != nil {
-		return nil, common.NewApiError(http.StatusInternalServerError, err)
+		log.WithError(err).Errorf("failed to set install config overrides feature usage for cluster %s", params.ClusterID)
```

Comment:
do we not return here on purpose?

Reply:
Yes, a failure to set the feature usage isn't a failure to update the install config override so we only print the error instead of returning it. Should it return an error instead?

Reply:
Yeah that's what I thought, OK 

> Should it return an error instead?

No, but it would be nice to add a comment where a return would usually be to explain this, so no one would get confused by this in the future and accidentally even put one in there to "fix" it

Reply:
Gotcha! Noted for the future too, thank you Omer!